instrumentation/micrometer/micrometer-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/OpenTelemetryCounter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.micrometer.v1_5;
+
+import static io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.Bridging.baseUnit;
+import static io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.Bridging.description;
+import static io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.Bridging.toAttributes;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Measurement;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleCounter;
+import io.opentelemetry.api.metrics.Meter;
+import java.util.Collections;
+
+final class OpenTelemetryCounter implements Counter, RemovableMeter {
+
+  private final Id id;
+  // TODO: use bound instruments when they're available
+  private final DoubleCounter otelCounter;
+  private final Attributes attributes;
+
+  volatile boolean removed = false;
+
+  OpenTelemetryCounter(Id id, Meter otelMeter) {
+    this.id = id;
+    this.otelCounter =
+        otelMeter
+            .counterBuilder(id.getName())
+            .setDescription(description(id))
+            .setUnit(baseUnit(id))
+            .ofDoubles()
+            .build();
+    this.attributes = toAttributes(id.getTags());
+  }
+
+  @Override
+  public void increment(double v) {
+    if (removed) {
+      return;
+    }
+    otelCounter.add(v, attributes);
+  }
+
+  @Override
+  public double count() {
+    // OpenTelemetry metrics bridge does not support reading measurements
+    return Double.NaN;
```

Comment:
Do we know how this is commonly used?

Reply:
I think it's only supposed to be used in the `measure()` method - which in turn is only called in actual `MeterRegistry` implementations (like the Signalfx one, for example). But since this is public API anyone can call it -- I don't really know why though, it's not particularly useful.
The `Timer` is a bit of an exception to that - its `count()`, `totalTime()` and `max()` methods are also called to compute the histogram.

> should we log a one-time warning/debug in case it's being used?

👍